Documentation/exposing-prometheus-and-alertmanager.md

```diff
@@ -0,0 +1,284 @@
+# Exposing Prometheus and Alertmanager
+
+The Prometheus Operator takes care of operating Prometheus and Alertmanagers
+clusters, however, there are many ways in Kubernetes to expose these to the
+outside world. This document shall outline best practices and caveats to do so
+in various ways.
+
+## NodePort
+
+While being the least practical solution in a production environment the
+easiest solution is to create a `Service` for a Prometheus cluster.
+
+A simple Prometheus object's manifest could look like this:
+
+```
+apiVersion: "monitoring.coreos.com/v1alpha1"
+kind: "Prometheus"
+metadata:
+  name: "main"
+spec:
+  replicas: 1
+  version: v1.5.0
+  resources:
+    requests:
+      memory: 400Mi
+```
+
+All Prometheus `Pod`s are labelled with `prometheus: <prometheus-object-name>`,
+as the Prometheus object's name is `main`, the selector ends up being
+`prometheus: main`. Meaning, the respective manifest for the `Service` would
+look like this:
+
+```
+apiVersion: v1
+kind: Service
+metadata:
+  name: prometheus-main
+spec:
+  type: NodePort
+  ports:
+  - name: web
+    nodePort: 30900
+    port: 9090
+    protocol: TCP
+    targetPort: web
+  selector:
+    prometheus: main
+```
+
+And just by creating this `Service` the web UI of Prometheus will be accessible
+by browsing to any of the worker nodes using `http://node-ip:30900/`.
+
+Exposing the Alertmanager works in the same fashion, the only difference being,
+that the selector is `alertmanager: <alertmanager-object-name>`. So the
+`Alertmanager` manifest would look like this:
+
+```
+apiVersion: "monitoring.coreos.com/v1alpha1"
+kind: "Prometheus"
+metadata:
+  name: "main"
+spec:
+  replicas: 1
+  version: v1.5.0
+  resources:
+    requests:
+      memory: 400Mi
+```
+
+And the `Service` manifest like this:
+
+```
+apiVersion: v1
+kind: Service
+metadata:
+  name: alertmanager-main
+spec:
+  type: NodePort
+  ports:
+  - name: web
+    nodePort: 30903
+    port: 9093
+    protocol: TCP
+    targetPort: web
+  selector:
+    alertmanager: main
+```
+
+And the Alertmanager web UI will be available at `http://node-ip:30903/`.
+
+## kubectl proxy
+
+The Prometheus and Alertmanager web UIs may only be interesting to cluster
+admins, or people who access the cluster with `kubectl` regularly. In this case 
+it seems natural to access those using `kubectl proxy`.
+
+To be able to do so, create a `Service` of type `ClusterIP`.
+
+```
+apiVersion: v1
+kind: Service
+metadata:
+  name: prometheus-main
+spec:
+  type: ClusterIP
+  ports:
+  - name: web
+    port: 9090
+    protocol: TCP
+    targetPort: web
+  selector:
+    prometheus: main
+```
+
+Once we now create the Prometheus object and all `Pod`s have been created
+properly, and we execute
+
+```
+kubectl proxy
```

Comment:
This method is available on basically any exposed API endpoint, such as the one a Tectonic installation would offer. I'm thinking that describing this method relative to the API endpoint might appeal to a wider audience with more complex setups. Mentioning the `kubectl proxy` step as an optional extra for convenience should help people understand that it's only needed if they don't have an otherwise reachable API.